### PR TITLE
Add scope representing full user permissions, default token requests with unspecified scope

### PR DESCRIPTION
**Pull request by liggitt**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1359035

Default token authorization requests that do not specify a scope to a `user:full` scope which includes all of the user's powers.

- [ ] doc PR for new scope

### CODE REVIEW

**Comment by liggitt** [comment]
[test]

File: pkg/authorization/authorizer/scope/converter.go [hunk 1]
```diff
@@ -107,32 +108,38 @@ func (userEvaluator) Validate(scope string) error {
...
+	case UserInfo:
 		return "Information about you, including: username, identity names, and group membership."
-	case UserIndicator + UserAccessCheck:
+	case UserAccessCheck:
 		return `Information about user privileges, e.g. "Can I create builds?"`
-	case UserIndicator + UserListProject:
+	case UserListProject:
 		return `See projects you're aware of and the metadata (display name, description, etc) about those projects.`
+	case UserFull:
+		return `All permissions of the user.`
```
Comment (deads2k) [inline pkg/authorization/authorizer/scope/converter.go]: Someday we're going to need, "all non-escalating".  I can feel it.

Comment (deads2k) [inline pkg/authorization/authorizer/scope/converter.go]: Make this consistent.  Either "you/your" everywhere or "user" everywhere.

File: pkg/util/html/form.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+func GetAttr(element *html.Node, attrName string) (string, bool) {
+	for _, attr := range element.Attr {
+		if attr.Key == attrName {
+			return attr.Val, true
+		}
+	}
+	return "", false
+}
+
+func NewRequestFromForm(form *html.Node, currentURL *url.URL) (*http.Request, error) {
```
Comment (deads2k) [inline pkg/util/html/form.go]: I'm not normally a nit-picky kind of guy, but how about some doc about what the heck this doing.

File: pkg/authorization/authorizer/scope/converter.go [hunk 3]
```diff
@@ -107,32 +108,38 @@ func (userEvaluator) Validate(scope string) error {
...
 			authorizationapi.NewRule("create").Groups(authorizationapi.GroupName).Resources("selfsubjectrulesreviews").RuleOrDie(),
 		}, nil
-	case UserIndicator + UserListProject:
+	case UserListProject:
 		return []authorizationapi.PolicyRule{
 			{Verbs: sets.NewString("list"), APIGroups: []string{projectapi.GroupName}, Resources: sets.NewString("projects")},
 		}, nil
+	case UserFull:
+		return []authorizationapi.PolicyRule{
+			{Verbs: sets.NewString("*"), APIGroups: []string{"*"}, Resources: sets.NewString("*")},
```
Comment (deads2k) [inline pkg/authorization/authorizer/scope/converter.go]: Why aren't you allowing access to non-resource URLs?

File: pkg/util/html/form.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
+package html
```
Comment (deads2k) [inline pkg/util/html/form.go]: if this is only for integration testing, can we hide it down there?

**Comment by deads2k** [comment]
Other than the question about non-resource URLs, everything else is minor.

**Comment by liggitt** [comment]
comments addressed

**Comment by liggitt** [comment]
[merge]

**Comment by liggitt** [comment]
https://ci.openshift.redhat.com/jenkins/job/test_pr_origin/7712/ failed with bug #10319 and flakes https://github.com/openshift/origin/issues/9548 https://github.com/openshift/origin/issues/9490, re[merge]

**Pull request merged**